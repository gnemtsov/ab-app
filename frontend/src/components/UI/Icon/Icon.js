import React from 'react';
import feather from './feather-sprite.svg';

const Icon = (props) => {
    let defaults = {
        width: 24,
        height: 24,
        fill: 'none',
        stroke: 'currentColor',
        strokeWidth: 2,
        strokeLinecap: 'round',
        strokeLinejoin: 'round'
    }

    const params = {
        ...defaults,
        ...props
    }

    return (
        <svg {...params}>
            <use xlinkHref={feather + '#' + props.name} />
        </svg>
    );
}

export default Icon;