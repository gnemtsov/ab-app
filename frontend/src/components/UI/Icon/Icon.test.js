import React from 'react';

import { configure, render, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Icon from './Icon';

configure({ adapter: new Adapter() });

describe('Icon', () => {
	it('Renders correctly', () => {
		const wrapper = shallow(<Icon name="test" />);
		expect(wrapper).toMatchSnapshot();
	});
})