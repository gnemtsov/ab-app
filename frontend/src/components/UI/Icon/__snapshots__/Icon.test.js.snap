// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Renders correctly 1`] = `
<svg
  fill="none"
  height={24}
  name="test"
  stroke="currentColor"
  strokeLinecap="round"
  strokeLinejoin="round"
  strokeWidth={2}
  width={24}
>
  <use
    xlinkHref="test-pic-stub#test"
  />
</svg>
`;
